packages/grid/x-data-grid/src/hooks/features/columns/useGridColumns.tsx

```diff
@@ -232,17 +233,30 @@ export function useGridColumns(
     (field, width) => {
       logger.debug(`Updating column ${field} width to ${width}`);
 
-      const column = apiRef.current.getColumn(field);
-      const newColumn = { ...column, width };
-      apiRef.current.updateColumns([newColumn]);
```

Comment:
It doesn't make much sense to call `updateColumns` to resize them, so I updated the implementation to calculate the new columns' width and update the state.